nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/activations/impl/ActivationReLU.java

```diff
@@ -101,6 +104,33 @@ public INDArray getActivation(INDArray in, boolean training) {
         return new Pair<>(dLdz, null);
     }
 
+    @Override
+    public SDVariable defineActivation(@NonNull SameDiff sameDiff, @NonNull SDVariable input) {
+        SDVariable temp;
+        double thresh = threshold == null ? 0.0 : threshold;
+        double ns = negativeSlope == null ? 0.0 : negativeSlope;
+        if(ns == 0){
+            temp = sameDiff.nn.relu(input, thresh);
+        } else {
+            if(thresh == 0)
+                temp = sameDiff.nn.leakyRelu(input, negativeSlope);
+            else {
+                //TODO optimize this
+                SDVariable t = sameDiff.constant(thresh).castTo(input.dataType());
+                SDVariable oneGte = input.gte(t).castTo(input.dataType());
+                SDVariable oneLt = input.lt(t).castTo(input.dataType());
+                SDVariable lower = oneLt.mul(ns).mul(input.sub(threshold));
+                SDVariable upper = oneGte.mul(input);
+                temp = lower.add(upper);
+            }
+        }
+
+        if(max != null)
+            temp = sameDiff.math.max(sameDiff.constant(max).castTo(temp.dataType()), temp);
```

Comment:
Probably won't matter much given the inefficiency for the threshold case, but we can define 2 of these constants without the castTo (as per earlier comments)